### DISCUSSION TITLE
Schema generation and visualisation, + video

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#228] [General]
We recently added the `schema` app to Piccolo. It lets users auto generate their Piccolo `Table` classes from an existing database, using `piccolo schema generate`.

It also auto generates an Entity Relationship Diagram from your schema using `piccolo schema graph`, which can be great for documentation.

Here's a video about it:

https://www.youtube.com/watch?v=Y9vaS4nnQGE

**Comment by James-Firth** [comment]
I tried using `piccolo schema generate` today while seeing if piccolo would work as a nice query builder for one of our teams.

But we have a table called `user` that referenced a _lot_ so I got a big list of `ValueError`s with the exception `user is a protected name, please give your table a different name`.

Is there a way for me to tell the generator to ignore that error or use a different alias internally?

I know Piccolo does a lot, but I'm really just looking for something to do models and query building right now so I don't _think_ the `user` restriction should matter to me.

Thanks in advance and let me know if I should put this in a separate issue or discussion :)

**Reply by dantownsend** [reply]
Yeah, we originally added a check because `user` refers to the current logged in user, so have having a table called `user` is sometimes problematic.

https://github.com/piccolo-orm/piccolo/blob/29f4c2ffd5a9f297779898ca903224d4052a8c87/piccolo/table.py#L259-L263

However ... we're really careful about double quoting any table references so we probably don't need this anymore. I've been thinking of getting rid of it for a while. I'll have a look tonight - will remove it, or at the very least change it to a warning instead of raising an exception.

**Reply by dantownsend** [reply]
You can try commenting out those lines for now.

**Reply by dantownsend** [reply]
I'll create an issue.

**Reply by dantownsend** [reply]
https://github.com/piccolo-orm/piccolo/issues/849

**Reply by dantownsend** [reply]
@James-Firth Try `piccolo==0.116.0` - it now just emits a warning instead of an exception. I might remove this warning in the future, but at least it should work now.

**Reply by James-Firth** [reply]
Thanks! Just gave that a shot and it looks like it works, thanks!

**Reply by dantownsend** [reply]
Nice!
